File: backend/schemas/suggestion.py
from enum import StrEnum
from typing import Annotated

from pydantic import BaseModel, Field


class SuggestionState(StrEnum):
    IDLE = "idle"
    LOADING = "loading"
    SUCCESS = "success"
    ERROR = "error"


class SuggestionRecord(BaseModel):
    score: Annotated[
        int,
        Field(
            description="The score of the suggestion",
            ge=0,
            le=100,
        ),
    ]
    purpose: Annotated[
        str,
        Field(description="The purpose of the suggestion"),
    ]
    content: Annotated[
        str,
        Field(description="The content of the suggestion"),
    ]


class SuggestionBatch(BaseModel):
    timestamp: Annotated[
        int,
        Field(description="The Unix timestamp of the suggestions"),
    ]
    last_question: Annotated[
        str,
        Field(description="The last question"),
    ]
    suggestions: Annotated[
        list[SuggestionRecord],
        Field(description="The list of suggestions"),
    ]

File: ui/types/suggestion.ts
export enum SuggestionState {
  IDLE = 'idle',
  PENDING = 'pending',
  LOADING = 'loading',
  SUCCESS = 'success',
  STOPPED = 'stopped',
  ERROR = 'error',
}

export interface ReplySuggestion {
  timestamp: number;
  last_question: string;
  answer: string;
  state: SuggestionState;
}

export interface CodeSuggestion {
  timestamp: number;
  image_urls: string[];
  user_prompt: string | null;
  suggestion_content: string;
  state: SuggestionState;
}
